File: code/greedy_algorithms/src/egyptian_fraction/egyptian_fraction.php

<?php 
// Part of Cosmos by OpenGenus
// PHP program to print a fraction  
// in Egyptian Form using Greedy 
// Algorithm 
  
function printEgyptian($nr, $dr) 
{ 
    // If either numerator or  
    // denominator is 0 
    if ($dr == 0 || $nr == 0) 
        return; 
  
    // If numerator divides denominator,  
    // then simple division makes the  
    // fraction in 1/n form 
    if ($dr % $nr == 0) 
    { 
        echo "1/", (int)$dr / $nr; 
        return; 
    } 
  
    // If denominator divides numerator,  
    // then the given number is not fraction 
    if ($nr%$dr == 0) 
    { 
        echo (int)($nr / $dr); 
        return; 
    } 
  
    // If numerator is more than denominator 
    if ($nr > $dr) 
    { 
        echo (int)($nr/$dr), " + "; 
        printEgyptian($nr % $dr, $dr); 
        return; 
    } 
  
    // We reach here dr > nr and dr%nr is  
    // non-zero. Find ceiling of dr/nr and  
    // print it as first fraction 
    $n = (int)($dr / $nr ) + 1; 
    echo "1/" , $n , " + "; 
  
    // Recur for remaining part 
    printEgyptian($nr * $n - $dr, $dr * $n); 
} 
  
// Driver Code 
$nr = 6; 
$dr = 14; 
echo "Egyptian Fraction Representation of ", 
                    $nr, "/", $dr, " is\n "; 
printEgyptian($nr, $dr); 

?> 

File: code/greedy_algorithms/src/egyptian_fraction/egyptian_fraction.java
class GFG { 
  
    static void printEgyptian(int nr, int dr) { 
        // If either numerator or  
        // denominator is 0  
        if (dr == 0 || nr == 0) { 
            return; 
        } 
  
        // If numerator divides denominator,  
        // then simple division makes  
        // the fraction in 1/n form  
        if (dr % nr == 0) { 
            System.out.print("1/" + dr / nr); 
            return; 
        } 
  
        // If denominator divides numerator,  
        // then the given number is not fraction  
        if (nr % dr == 0) { 
            System.out.print(nr / dr); 
            return; 
        } 
  
        // If numerator is more than denominator  
        if (nr > dr) { 
            System.out.print(nr / dr + " + "); 
            printEgyptian(nr % dr, dr); 
            return; 
        } 
  
        // We reach here dr > nr and dr%nr  
        // is non-zero. Find ceiling of  
        // dr/nr and print it as first  
        // fraction  
        int n = dr / nr + 1; 
        System.out.print("1/" + n + " + "); 
  
        // Recur for remaining part  
        printEgyptian(nr * n - dr, dr * n); 
    } 
  
// Driver Code  
    public static void main(String[] args) { 
        int nr = 6, dr = 14; 
        System.out.print("Egyptian Fraction Representation of "
                + nr + "/" + dr + " is\n "); 
        printEgyptian(nr, dr); 
    } 
} 
